snuba/datasets/dataset.py

```diff
@@ -63,9 +63,7 @@ def get_query_plan_builder(
         # TODO: If the query is being executed on a single entity, that entity will be used to determine
         # the query plan. In cases such as a join, the dataset will something something and
         # then build the join query.
-        entity = self.get_entity(entity_name)
-
-        return entity.get_query_plan_builder()
+        return self.__default_entity.get_query_plan_builder()
```

Comment:
Eventually I think that `Dataset.get_query_plan_builder()` should be deprecated and replaced everywhere with `Entity.get_query_plan_builder()` so we don't have this issue and we can run different query plans for different entities in the same dataset. I haven't tackled this here since it probably requires quite a few more changes, so instead I just added a temporary method to switch the default entity during the entity selection - let me know if this makes sense to you

Reply:
This would work for queries that involve one entity at a time. Though for joins we need something more complex in that each entity can pick its own storage but there needs to be an overall query plan and an overall plan builder to ensure compatible storages are picked.